### PR TITLE
RFXCOM Add binding support for 10 sensor types

### PR DESCRIPTION
**Pull request by Speederc**
Add binding support for the following sensor types :
- CurrentEnergy (tested in #3854)
- DateTime (*)
- Fan (*)
- Lighting3 (*)
- Power (*)
- TemperatureHumidityBarometric (tested in #2557)
- TemperatureRain (tested in #3863)
- Thermostat2 (*)
- Thermostat3 (*)
- Weight (*)

 (*) Untested sensor. Can be buggy ! But at least it gives a first draft for future fix.

### CODE REVIEW

**Comment by ivanfmartinez** [comment]
The implementation apper to be ok, except for the comments I made directly in code and the other s that are already corrected by [https://github.com/openhab/openhab/pull/3957](url)

I have one suggestion to be usedt abou data lengths as a pattern  for this messages :  
    byte[] data = new byte[TOTAL_LENGTH];
   data[0] = data.length-1;

Only people with correct devices can test the working condition.

**Comment by Speederc** [comment]
Thanks for your comments, I will update this PR with the fixes.

**Comment by Speederc** [comment]
Done ! Can be merged.

File: bundles/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/messages/RFXComCurrentEnergyMessage.java [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+import org.openhab.binding.rfxcom.internal.RFXComException;
+import org.openhab.core.library.items.NumberItem;
+import org.openhab.core.library.items.StringItem;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.Type;
+import org.openhab.core.types.UnDefType;
+
+public class RFXComCurrentEnergyMessage extends RFXComBaseMessage {
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/messages/RFXComCurrentEnergyMessage.java]: please add javadoc on class level

File: bundles/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/messages/RFXComPowerMessage.java [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+import org.openhab.binding.rfxcom.internal.RFXComException;
+import org.openhab.core.library.items.NumberItem;
+import org.openhab.core.library.items.StringItem;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.Type;
+import org.openhab.core.types.UnDefType;
+
+public class RFXComPowerMessage extends RFXComBaseMessage {
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/messages/RFXComPowerMessage.java]: please add javadoc on classlevel

**Comment by teichsta** [comment]
please find some minor comments inline. Some classes appear to be complete new although they were available already? What happened to them in your PR? Reformatting?

**Comment by Speederc** [comment]
I fixed the minor comments. What do you mean by "Some classes appear to be complete new although they were available already" ? All these new types of messages were unsupported until this PR.

**Comment by teichsta** [comment]
Thanks @Speederc!

**Comment by martinvw** [comment]
@Speederc did you have plans to add this support to the binding of OH2?

**Comment by martinvw** [comment]
I will make start on it :-)

**Comment by martinvw** [comment]
@Speederc I ported the first of the messages, but @kaikreuzer requested me to add Also-By: line to indicate you agree to the [the DCO](https://github.com/openhab/openhab2-addons/blob/master/CONTRIBUTING.md#sign-your-work)

Since you did not sign-off this commit, do you agree to the DCO?

**Comment by Speederc** [comment]
Hi @martinvw , sorry I don't have so much time yet for Openhab. Thus I won't be able to add this support in OH2. Thank you for doing it !
Of course I agree to the DCO.

**Comment by martinvw** [comment]
I just got a request to specifically port the TemperatureRain message, which I worked on, see https://community.openhab.org/t/rfxcom-binding-does-not-decode-temperature-rain-alecto-ws-1200-message/6484/6

However if other people are in need for a specific one, just give a shout. It takes me approximately half an hour for a single message type, so I can do it quite quickly. And you can help me by supplying some example data, in that way I can add a proper test.

**Pull request merged**